custom_components/adaptive_lighting/switch.py

```diff
@@ -747,6 +823,13 @@ def __init__(
         self._icon = ICON_MAIN
         self._state = None
 
+        self._transition_timer = 0
+        self._transitioning = False
+        self._manual_lights = {}
+        all_lights = _expand_light_groups(self.hass, self._lights)
+        for light in all_lights:
+            self._manual_lights[light] = {"timer": 0}
```

Comment:
Maybe add a debugging log statement here to see whether the lights are really in the dict?

Reply:
I fixed this with @nathang21 last time, the lights somehow weren't initialized there for light groups so I had to readd the check in `AdaptiveSwitch._expand_light_groups`

I'm going to test on my own HASS system like I've been doing to see if anything is crazy on my home system.

Reply:
```
File "/config/custom_components/adaptive_lighting/switch.py", line 560, in _supported_features
supported_features = state.attributes[ATTR_SUPPORTED_FEATURES]
AttributeError: 'NoneType' object has no attribute 'attributes'
2023-03-26 21:06:14.351 ERROR (MainThread) [homeassistant.components.switch] Error adding entities for domain switch with platform adaptive_lighting
```
There we go, this is a very different error that I couldn't find in the mountain of other checks pytest was doing. The original error. Is this your code?

Reply:
Well I managed to reproduce that person's bug somehow.

Reply:
Here this means that `state = None`